Create docker-publish.yml

ZachNewbery opened the pull request:
Allow publishing of container to GHCR.io

ZachNewbery committed:
Create docker-publish.yml

Allow publishing of container to GHCR.io

fingolfin commented:
Thanks for your contribution!

Could you explain a bit what this does / is meant to achieve?

Googling a bit it seems to be closely based on the example at https://docs.github.com/en/packages/managing-github-packages-using-github-actions-workflows/publishing-and-installing-a-package-with-github-actions -- perhaps it'd be helpful to include that link so that people who need to edit this file in the future have a good starting point?

ZachNewbery commented:
Hiya, apologies for the late reply - exam season has been stressful! But yeah, essentially the script is trying to migrate the deployment of the docker containers for GAP from dockerhub to the free alternative of Github's own container repository. 

Right now it doesn't do much beyond the basic examples (https://github.com/actions/starter-workflows/blob/da223f8a03b845b8624b144879658659a0b97d76/ci/docker-publish.yml), and deploys the container every day at 7am. I have been looking at possibly getting it to run some tests as well using this example (https://github.com/docker/build-push-action)

ZachNewbery committed:
Remove Manuals download from Dockerfile

Currently, there is no tar ball associated with the following link for GAP 4.10.2:
https://www.gap-system.org/Manuals/gap-4.10.2-manuals.tar.gz
and as such the whole image deployment fails. This commit removes those instructions to allow for deployment of the Docker image.

ZachNewbery committed:
Merge pull request #1 from ZachNewbery/ZachNewbery-patch-1

Currently, there is no tar ball associated with the following link for GAP 4.10.2:
https://www.gap-system.org/Manuals/gap-4.10.2-manuals.tar.gz
and as such the whole image deployment fails. This commit removes those instructions to allow for deployment of the Docker image.

While this allows the dockerfile to be migrated successfully to GHCR.io, it should be something to be aware of for later versions of GAP (i.e. is only a temporary fix).
